pkg/authorization/registry/clusterrole/proxy/proxy.go

```diff
@@ -1,124 +1,150 @@
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	apirequest "k8s.io/apiserver/pkg/endpoints/request"
 	"k8s.io/apiserver/pkg/registry/rest"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/rbac/internalversion"
 
-	authorizationapi "github.com/openshift/origin/pkg/authorization/apis/authorization"
-	clusterpolicyregistry "github.com/openshift/origin/pkg/authorization/registry/clusterpolicy"
-	"github.com/openshift/origin/pkg/authorization/registry/clusterrole"
-	roleregistry "github.com/openshift/origin/pkg/authorization/registry/role"
-	rolestorage "github.com/openshift/origin/pkg/authorization/registry/role/policybased"
-	"github.com/openshift/origin/pkg/authorization/rulevalidation"
+	authzapi "github.com/openshift/origin/pkg/authorization/apis/authorization"
 )
 
-type ClusterRoleStorage struct {
-	roleStorage rolestorage.VirtualStorage
+func rbacToClusterRole(in *rbac.ClusterRole) (authzapi.ClusterRole, error) {
+	var out authzapi.ClusterRole
+	err := authzapi.Convert_rbac_ClusterRole_To_authorization_ClusterRole(in, &out, nil)
+	return out, err
 }
 
-func NewClusterRoleStorage(clusterPolicyRegistry clusterpolicyregistry.Registry, liveRuleResolver, cachedRuleResolver rulevalidation.AuthorizationRuleResolver) clusterrole.Storage {
-	return &ClusterRoleStorage{
-		roleStorage: rolestorage.VirtualStorage{
-			PolicyStorage: clusterpolicyregistry.NewSimulatedRegistry(clusterPolicyRegistry),
+func rbacFromClusterRole(in *authzapi.ClusterRole) (rbac.ClusterRole, error) {
+	var out rbac.ClusterRole
+	err := authzapi.Convert_authorization_ClusterRole_To_rbac_ClusterRole(in, &out, nil)
+	return out, err
+}
 
-			RuleResolver:       liveRuleResolver,
-			CachedRuleResolver: cachedRuleResolver,
+type ClusterRoleStorage struct {
+	client internalversion.ClusterRoleInterface
+}
 
-			CreateStrategy: roleregistry.ClusterStrategy,
-			UpdateStrategy: roleregistry.ClusterStrategy,
-			Resource:       authorizationapi.Resource("clusterrole"),
-		},
-	}
+func NewREST(client internalversion.ClusterRoleInterface) *ClusterRoleStorage {
+	return &ClusterRoleStorage{client}
 }
 
 func (s *ClusterRoleStorage) New() runtime.Object {
-	return &authorizationapi.ClusterRole{}
+	return &authzapi.ClusterRole{}
 }
 func (s *ClusterRoleStorage) NewList() runtime.Object {
-	return &authorizationapi.ClusterRoleList{}
+	return &authzapi.ClusterRoleList{}
 }
 
 func (s *ClusterRoleStorage) List(ctx apirequest.Context, options *metainternal.ListOptions) (runtime.Object, error) {
-	ret, err := s.roleStorage.List(ctx, options)
-	if ret == nil {
+	optv1 := metav1.ListOptions{}
+	if err := metainternal.Convert_internalversion_ListOptions_To_v1_ListOptions(options, &optv1, nil); err != nil {
 		return nil, err
 	}
-	return authorizationapi.ToClusterRoleList(ret.(*authorizationapi.RoleList)), err
+	roles, err := s.client.List(optv1)
```

Comment:
you definitely need to impersonate or you'll have disabled all the escalation checks.